example/convolution1D/src/convolution1D.cpp

```diff
@@ -0,0 +1,200 @@
+/* Copyright 2023 Benjamin Worpitz, Erik Zenker, Bernhard Manfred Gruber, Jan Stephan
+ * SPDX-License-Identifier: ISC
+ */
+
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+#include <type_traits>
+
+//! Convolution Example
+//!
+//! 1D convolution example: Creates two 1D arrays, applies convolution filter.
+//! Array sizes are hardcoded.
+//!
+
+/**
+ * @brief The ConvolutionKernel function-object
+ * Calculates 1D convolution using input and filter arrays.
+ */
+struct ConvolutionKernel
+{
+    /** @brief Main convolution code
+     *  @param Accelerator
+     *  @param Input array, first input of convolution integral
+     *  @param Filter array, second input of convolution integral
+     *  @param Empty output array to be filled
+     *  @param Input array size
+     *  @param Filter size
+     */
+    template<typename TAcc, typename TElem>
+    ALPAKA_FN_ACC auto operator()(
+        TAcc const& acc,
+        TElem const* const input,
+        TElem const* const filter,
+        TElem* const output,
+        const std::size_t inputSize,
+        const std::size_t filterSize) const -> void
+    {
+        using Idx = alpaka::Idx<TAcc>;
+        using Vec = alpaka::Vec<alpaka::Dim<TAcc>, Idx>;
+        using VecInt = alpaka::Vec<alpaka::DimInt<1u>, Idx>;
+
+
+        Vec const globalThreadIdx = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc);
+        Vec const globalThreadExtent = alpaka::getWorkDiv<alpaka::Grid, alpaka::Threads>(acc);
+
+        // Since the kernel is launched 1-D calculating linearizedGlobalThreadIdx line is unnecessary.
+        // globalThreadIdx[0] can be used to map all the threads.
+        if(globalThreadIdx[0] < static_cast<unsigned>(inputSize))
+        {
+            uint32_t const halfFilterSize = filterSize / 2;
+            TElem result = 0.0f;
+            // Calculate sum of multiplications of corresponding elements
+            for(size_t i = 0; i < filterSize; ++i)
+            {
+                uint32_t inputIndex = globalThreadIdx[0] - halfFilterSize + i;
+                if(inputIndex >= 0 && inputIndex < inputSize)
+                {
+                    result += input[inputIndex] * filter[i];
+                }
+            }
+            output[globalThreadIdx[0]] = result;
+        }
+    }
+};
+
+/**
+ * @brief Function template to compare floating point values
+ *
+ */
+template<typename FloatingType>
+typename std::enable_if<std::is_floating_point<FloatingType>::value, bool>::type FuzzyEqual(
+    FloatingType a,
+    FloatingType b)
+{
+    return std::fabs(a - b) < std::numeric_limits<FloatingType>::epsilon() * 10;
+}
```

Comment:
```suggestion
void FuzzyEqual(float a, float b)
{
    return std::fabs(a - b) < std::numeric_limits<float>::epsilon() * 10;
}
```
I would really simplify this to make the example less complex.

Reply:
I agree simpler should be better in the example.


```C++
auto FuzzyEqual(float a, float b) ...
```

Reply:
`10` should be a `10f`